window/src/os/x11/window.rs

```diff
@@ -872,6 +896,22 @@ impl XWindowInner {
         }
     }
 
+    fn set_text_cursor_position(&mut self, cursor: Rect) {
+        self.last_cursor_position = cursor;
```

Comment:
re: discussion about laggy input and IME positioning being expensive, perhaps a simple:

```rust
   if self.last_cursor_position == cursor {
       return;
   }
```

will help with this!

Reply:
Yes, I think this does the trick!